### PR TITLE
Feature/auth guards

### PR DESCRIPTION
**Pull request by MihaiVanuta**

### CODE REVIEW

**Review by cumpatacostin — APPROVED** [review]

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 import { LoginComponent } from './pages/login/login.component';
+import { AdminGuard } from 'src/guards/admin.guard';
```
Comment (denisx304) [inline src/app/app-routing.module.ts:4]: NIT: not needed anymore

File: src/app/pages/login/login.component.ts [hunk 2]
```diff
@@ -34,9 +39,37 @@ export class LoginComponent implements OnInit {
...
+              this.router.navigate(['user'])
+            }
+            else if(role === 'admin'){
+              this.router.navigate(['admin'])
+            }
+          });
+        }
+      },
+      error => {
+        console.log(error);
```
Comment (denisx304) [inline src/app/pages/login/login.component.ts:72]: User should be notified here.

File: src/app/services/auth/token-storage.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  public saveAuthorities(authorities: string[]) {
+    window.sessionStorage.removeItem(AUTHORITIES_KEY);
+    window.sessionStorage.setItem(AUTHORITIES_KEY, JSON.stringify(authorities));
+  }
+ 
+  public getAuthorities(): string[] {
+    this.roles = [];
+ 
+    if (sessionStorage.getItem(TOKEN_KEY)) {
+      JSON.parse(sessionStorage.getItem(AUTHORITIES_KEY)).forEach(authority => {
```
Comment (denisx304) [inline src/app/services/auth/token-storage.service.ts:45]: Always wrap methods that can throw an exception in a `try catch` block. 
`JSON.parse` can throw a SyntaxError exception. (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse)

File: src/app/interceptor/auth.interceptor.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        let authReq = req;
+        const token = this.token.getToken();
+        if (token != null) {
+            authReq = req.clone({ headers: req.headers.set(TOKEN_HEADER_KEY, 'Bearer ' + token) });
+        }
+        return next.handle(authReq);
+    }
+}
+ 
+export const httpInterceptorProviders = [
```
Comment (denisx304) [inline src/app/interceptor/auth.interceptor.ts:24]: This should be imported in the logged-in module.

**Pull request merged**